File: dushan-admin-backend/framework/starter_data_permission/definitions/enums/data_scope.py
from framework.common.enums.base_enum import BaseEnum


class DataScope(BaseEnum):
    """角色范围编码；多个角色取并集，有有效角色时包含本人。"""

    ALL = (1, "全部数据权限")
    DEPT_CUSTOM = (2, "指定部门数据权限")
    DEPT_ONLY = (3, "部门数据权限")
    DEPT_AND_CHILD = (4, "部门及以下数据权限")
    SELF = (5, "仅本人数据权限")

File: dushan-admin-frontend/apps/web-ele/src/constants/data-scope.ts
/**
 * 后端 PermissionDataScopeEnum 数据权限范围（代码常量）。
 * 与 `system_data_scope` 字典值 1-5 对应；DEPT_CUSTOM=2 时需要选择部门集合。
 */
export const DataScope = {
  ALL: 1,
  DEPT_AND_CHILD: 4,
  DEPT_CUSTOM: 2,
  DEPT_ONLY: 3,
  SELF: 5,
} as const;

export type DataScopeValue = (typeof DataScope)[keyof typeof DataScope];
